File: C#/Search in a Binary Search Tree.cs
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     public int val;
 *     public TreeNode left;
 *     public TreeNode right;
 *     public TreeNode(int val=0, TreeNode left=null, TreeNode right=null) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
public class Solution700 {
    public TreeNode SearchBST(TreeNode root, int val) {
        
        if(root==null) return root;
        if(root.val==val)
        {
            return root;
        }
        else
        {
            return val < root.val ? SearchBST(root.left,val):SearchBST(root.right,val);
        }

        /*
        iteration

        while(root != null && root.val != val){
            root = val<root.val? root.left:root.right;
        }
        return root;

        */
        
    }
}

File: Golang/Search in a Binary Search Tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func searchBST(root *TreeNode, val int) *TreeNode {
    if root != nil && val < root.Val {
        return searchBST(root.Left, val)
    } else if root != nil && val > root.Val {
        return searchBST(root.Right, val)
    } else {
        return root
    }
}